source/fundamentals/logging.txt

```diff
@@ -0,0 +1,298 @@
+           <version>1.7.30</version>
+         </dependency>
+         <!-- This package contains an SLF4J-Logback binding -->
+         <dependency>
+           <groupId>ch.qos.logback</groupId>
+           <artifactId>logback-classic</artifactId>
+           <version>1.2.3</version>
+         </dependency>
+         <!-- Logback core functionality -->
+         <dependency>
+           <groupId>ch.qos.logback</groupId>
+           <artifactId>logback-core</artifactId>
+           <version>1.2.3</version>
+         </dependency>
+
+      Once you have installed the above dependencies, connect to your
+      MongoDB instance and retrieve a document with the following code.
+
+      .. include:: /includes/fundamentals/logging.rst
+
+      You should see output that resembles the following:
+
+      .. code-block:: none
+
+         ...
+         1156 [main] DEBUG o.m.driver.protocol.command - Sending command '{"find": "<your collection>", "filter": {}, "limit": 1, "singleBatch": true, "$db": "<your db>", "lsid": {"id": {"$binary": {"base64": "<server session>", "subType": "04"}}}}' with request id 13 to database <your db> on connection <your connection> to server <your server>
+         1173 [main] DEBUG o.m.driver.protocol.command - Execution of command with request id 13 completed successfully in 30.01 ms on connection <your connection> to server <your server>
+   
+      For more information on Logback, see the
+      `Logback manual <http://logback.qos.ch/manual/>`__.
+   
+   .. tab:: Log4j2
+      :tabid: Log4j2-binding
+   
+      Add the following dependencies to your ``pom.xml`` file.
+
+      .. code-block:: xml      
+      
+         <!-- SLF4J API -->
+         <dependency>
+           <groupId>org.slf4j</groupId>
+           <artifactId>slf4j-api</artifactId>
+           <version>1.7.30</version>
+         </dependency>
+         <!-- SLF4J-Log4j2 binding -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-slf4j-impl</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+         <!-- Log4j2 core functionality -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-core</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+
+      Once you have installed the above dependencies, log an error using the
+      following code.
+
+      .. code-block:: java
+
+         import org.slf4j.Logger;
+         import org.slf4j.LoggerFactory;
+
+         ...
+
+         Logger logger = LoggerFactory.getLogger(App.class);
+         logger.error("Logging an Error");
+
+      You should see output that resembles the following:
+
+      .. code-block:: none
+
+         12:35:00.438 [main] ERROR <my package path> - Logging an Error
+
+      For more information on Log4j2, see the
+      `Log4j2 manual <https://logging.apache.org/log4j/2.x/manual/>`__.
+
+Configure Your Logger
```

Comment:
I really like that we have configuration examples here, but I think we also need to document the logger names that are actually used by the driver, as in the current reference docs (http://mongodb.github.io/mongo-java-driver/4.2/driver/reference/logging/).  Perhaps we can include that information somewhere above, and then use one of the logger names the driver uses in the examples below?

For example, you could configure the "org.mongodb.driver.connection" logger to DEBUG and show the increased logging messages for connection usage.

Reply:
Thank you for the note. Good catch, I apologies for not documenting the logger names. I agree that it is important that the logger names be included in the docs, as there is not a way for learners to easily get that information outside the documentation. I have added a "Logger Names" section where I give some background on logger names and how they are used in the driver.

Logger Section: